pkg/net/ticker.go

```diff
@@ -0,0 +1,46 @@
+package net
+
+import (
+	"sync"
+	"time"
+)
+
+type backoff struct {
+	current time.Duration
+	min     time.Duration
+	max     time.Duration
+
+	factor   int64
+	isSteady bool
+	mu       sync.Mutex
+}
+
+func newBackoff(min, max time.Duration, factor int64) *backoff {
+	return &backoff{min: min, max: max, current: min, factor: factor, isSteady: false}
+}
+
+func (b *backoff) Jitter(time.Duration) time.Duration {
+	b.mu.Lock()
+	b.current = minDuration(time.Duration(b.current.Nanoseconds()*b.factor), b.max)
```

Comment:
I think we should replace the `mu`/`current` pair with an `atomic.Value`.

It won't really matter in terms of performance, but I actually prefer atomics over locks.  They're easier to read, and they avoid silly mistakes like forgetting to release a lock.

Reply:
Okay, I also prefer it

Reply:
The loss of types is a fair point, but I usually address this by declaring a new type:
```go
type atomicDuration atomic.Duration

func(ad *atomic.Duration) Load() time.Duration {
    if d, ok := (*atomic.Value)(ad).Load().(time.Duration); ok {
        return d
    }
    
    return 0 // d == nil
}

// todo:  equivalent method for Store(d)
```

I don't think the lost update problem applies here, since there's only one writer (the the select loop).  But if you have any doubts, we can leave the mutex.

Reply:
But, there are two writers right? The Gossip loop calling `Reset`, and the Ticker loop calling `Jitter`. So, with the Lost Update problem the `Reset` or `Jitter` call may not take place. Or am I losing something here?